File: cpp/Node.cpp
#include <string>
using namespace std;
class Node {
    public:
        int data;
        int freq;
        string code;
        Node* left;
        Node* right;

        Node(int data, int freq) {
            this->data = data;
            this->freq = freq;
            this->code = "";
            this->left = nullptr;
            this->right = nullptr;
        }

        bool isLeaf() {
            return !left && !right;
        }

        void updateCode(string addedChar) {
            this->code = addedChar + this->code;
            if(this->left) this->left->updateCode(addedChar);
            if(this->right) this->right->updateCode(addedChar);
        }
};

File: python/Node.py
class Node:
    def __init__(self, data, freq):
        self.data = data
        self.freq = freq
        self.code = ""
        self.left = None
        self.right = None

    def isLeaf(self):
        return not self.left and not self.right

    def updateCode(self, addedChar):
        self.code = addedChar + self.code
        if self.left:
            self.left.updateCode(addedChar)
        if self.right:
            self.right.updateCode(addedChar)
